examples/c/ocp_qp.c

```diff
@@ -56,65 +53,67 @@ int main() {
     double x0[] = {1, 1};
     int idxb0[] = {0, 1};
 
+
     ocp_qp_solver_plan plan;
-    plan.qp_solver = FULL_CONDENSING_HPIPM;
-
-    ocp_qp_xcond_solver_config *config = ocp_qp_config_create(plan);
-
-    ocp_qp_dims dims;
-
-    int nx[] = {2, 2, 2, 2, 2, 2};
-    int nu[] = {1, 1, 1, 1, 1, 0};
-    int nb[] = {2, 0, 0, 0, 0, 0};
-    int ng[] = {0, 0, 0, 0, 0, 0};
-    int ns[] = {0, 0, 0, 0, 0, 0};
-    int nsbx[] = {0, 0, 0, 0, 0, 0};
-    int nsbu[] = {0, 0, 0, 0, 0, 0};
-    int nsg[] = {0, 0, 0, 0, 0, 0};
-    int nbx[] = {2, 0, 0, 0, 0, 0};
-    int nbu[] = {0, 0, 0, 0, 0, 0};
-
-    dims.N = QP_HORIZON;
-    dims.nx = nx;
-    dims.nu = nu;
-    dims.nb = nb;
-    dims.ng = ng;
-    dims.ns = ns;
-    dims.nsbx = nsbx;
-    dims.nsbu = nsbu;
-    dims.nsg = nsg;
-    dims.nbx = nbx;
-    dims.nbu = nbu;
-
-    ocp_qp_in *qp_in = ocp_qp_in_create(config, &dims);
-
-    double *hA[] = {A, A, A, A, A};
-    double *hB[] = {B, B, B, B, B};
-    double *hb[] = {b, b, b, b, b};
-    double *hQ[] = {Q, Q, Q, Q, Q, Q};
-    double *hS[] = {S, S, S, S, S, S};
-    double *hR[] = {R, R, R, R, R};
-    double *hq[] = {q, q, q, q, q, q};
-    double *hr[] = {r, r, r, r, r, r};
-    int *hidxbx[] = {idxb0};
-    double *hlbx[] = {x0};
-    double *hubx[] = {x0};
-
-    d_ocp_qp_set_all(hA, hB, hb, hQ, hS, hR, hq, hr, hidxbx, hlbx, hubx,
-                     NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL,
-                     NULL, NULL, NULL, NULL, NULL, qp_in);
+    plan.qp_solver = FULL_CONDENSING_HPIPM; // FULL_CONDENSING_QPOASES, FULL_CONDENSING_HPIPM
+
+    ocp_qp_xcond_solver_config *config = ocp_qp_xcond_solver_config_create(plan);
+
+    int N = QP_HORIZON;
+    ocp_qp_xcond_solver_dims *dims = ocp_qp_xcond_solver_dims_create(config, N);
+
+    int nx = 2;
+    int nu = 1;
+    int nu_e = 0;
+    // here: no general linear constraints (ng), soft constraints (ns, nsbx, nsbu, nsg)
+
+    for (int i = 0; i < N+1; i++)
+    {
+        ocp_qp_xcond_solver_dims_set(config, dims, i, "nx", &nx);
+        ocp_qp_xcond_solver_dims_set(config, dims, i, "nu", &nu);
+    }
+    // last stage has no controls
+    ocp_qp_xcond_solver_dims_set(config, dims, N+1, "nu", &nu_e);
+
+    ocp_qp_xcond_solver_dims_set(config, dims, 0, "nbx", &nx);
+
+    ocp_qp_in *qp_in = ocp_qp_in_create_from_xcond_dims(dims);
```

Comment:
I wouldn't go for this.

Here IMO the correct thing to do is to create a `qp_dims` of type `ocp_qp_dims` to set the QP up, and a different `solver_dims` of type `ocp_qp_xcond_solver_dims` to set the solver up.

So there is no need for this ` ocp_qp_in_create_from_xcond_dims`.
At most, there could be a routine to conveniently fill in the `solvers_dims` by using a previously-created `qp_dims`.
This routine could internally use this HPIPM routine
https://github.com/giaf/hpipm/blob/master/ocp_qp/x_ocp_qp_dim.c#L148
in order to assure maintainability in case of changes in HPIPM.